python/ray/autoscaler/local/node_provider.py

```diff
@@ -4,11 +4,16 @@
 import os
 import socket
 import logging
+from http.client import RemoteDisconnected
```

Comment:
Add as a file comment, that this file contains two different types of node providers, and explain the differences.

Reply:
I added the comment. Though, now it seems to me that the two node providers do not necessarily have to be in the same node_provider.py file as the CoordinatorSenderNodeProvider can hook up with any "remote" node provider and the fact that we currently use it only with private/local clusters does not mean that it is the only use case. What do you think?

Reply:
Makes sense to have two files.